test/unit/services/test_catalogue_category.py

```diff
@@ -19,1167 +28,1047 @@
+            assert isinstance(actual_catalogue_category_in, CatalogueCategoryIn)
+            assert actual_catalogue_category_in.model_dump() == {
+                **self._expected_catalogue_category_in.model_dump(),
+                "properties": [
+                    {**prop.model_dump(), "id": ANY} for prop in self._expected_catalogue_category_in.properties
+                ],
+            }
+        else:
+            self.mock_catalogue_category_repository.create.assert_called_once_with(self._expected_catalogue_category_in)
 
-def test_delete(catalogue_category_repository_mock, catalogue_category_service):
-    """
-    Test deleting a catalogue category.
+        assert self._created_catalogue_category == self._expected_catalogue_category_out
 
-    Verify that the `delete` method properly handles the deletion of a catalogue category by ID.
-    """
-    catalogue_category_id = str(ObjectId())
+    def check_create_failed_with_exception(self, message: str):
+        """Checks that a prior call to `call_create_expecting_error` worked as expected, raising an exception
+        with the correct message"""
 
-    catalogue_category_service.delete(catalogue_category_id)
+        self.mock_catalogue_category_repository.create.assert_not_called()
 
-    catalogue_category_repository_mock.delete.assert_called_once_with(catalogue_category_id)
+        assert str(self._create_exception.value) == message
 
 
-def test_get(test_helpers, catalogue_category_repository_mock, catalogue_category_service):
-    """
-    Test getting a catalogue category.
+class TestCreate(CreateDSL):
+    """Tests for creating a catalogue category"""
 
-    Verify that the `get` method properly handles the retrieval of a catalogue category by ID.
-    """
-    # pylint: disable=duplicate-code
-    catalogue_category_id = str(ObjectId())
-    catalogue_category = MagicMock()
+    def test_create_without_properties(self):
+        """Test creating a catalogue category without properties"""
 
-    # Mock `get` to return a catalogue category
-    test_helpers.mock_get(catalogue_category_repository_mock, catalogue_category)
+        self.mock_create(CATALOGUE_CATEGORY_POST_DATA_NON_LEAF_NO_PARENT_NO_PROPERTIES_A)
+        self.call_create()
+        self.check_create_success()
 
-    retrieved_catalogue_category = catalogue_category_service.get(catalogue_category_id)
+    def test_create_with_properties(self):
+        """Test creating a catalogue category with properties"""
 
-    catalogue_category_repository_mock.get.assert_called_once_with(catalogue_category_id)
-    assert retrieved_catalogue_category == catalogue_category
+        self.mock_create(CATALOGUE_CATEGORY_DATA_LEAF_NO_PARENT_WITH_PROPERTIES_MM, units_in_data=[UNIT_IN_DATA_MM])
+        self.call_create()
+        self.check_create_success()
 
+    def test_create_with_duplicate_properties(self):
+        """Test creating a catalogue category with properties"""
```

Comment:
Not 100% sure this test is necessary - its handled by the `utils` so its kind of testing nested code here, while the function call is already asserted when there are properties anyway. Perhaps its a good idea to ensure the interaction is correct e.g. it raising an error doesn't break some further logic, but then again the e2e should cover the interaction part anyway.

Reply:
The same is the case for the update - I have not added that one back though, if we want this I will add that one back in.

Reply:
If it is handled by the `utils` then we probably do not need it.